store/storetest/status_store.go

```diff
@@ -26,7 +26,11 @@ func testStatusStore(t *testing.T, ss store.Store) {
 
 	status.LastActivityAt = 10
 
-	if err := (<-ss.Status().Get(status.UserId)).Err; err != nil {
+	if err := ss.Status().SaveOrUpdate(status); err != nil {
```

Comment:
You are adding code here that wasn't there before.

Reply:
The reason I did this is GoLand (the IDE I am using) alerted me that it was an invalid operation ("receive from non-chan type (*model.Status, *model.AppError)") which seemed to make sense since the StatusStore interface in store.go has (*model.Status, *model.AppError) as its return type and this portion of the test failed when run locally as well due to a similar reason ("multiple-value ss.Status().Get() in single-value context"). If this change is undesirable I will find a way to fix without altering the line mentioned here

Reply:
The problem that GoLand was complaining about was because previously we had `(<-ss.Status().Get(status.UserId)).Err`, which is, read the channel from the response that comes from `Status().Get(status.UserId)` that is no longer a channel, so you have to change the code as you did a bit later, the thing is, I don't see how this: `if err := ss.Status().SaveOrUpdate(status); err != nil {` is related to the problem that you have explained.

Reply:
Upon further review (and getting some sleep) understood your original complaint and noticed my mistake, my apologies.

Reply:
No problem :)